scene/main/viewport.cpp

```diff
@@ -3654,7 +3671,7 @@ Control *Viewport::gui_get_focus_owner() const {
 
 Control *Viewport::gui_get_hovered_control() const {
 	ERR_READ_THREAD_GUARD_V(nullptr);
-	return gui.mouse_over;
+	return Object::cast_to<Control>(ObjectDB::get_instance(gui.mouse_over));
```

Comment:
```suggestion
	return ObjectDB::get_instance<Control>(gui.mouse_over);
```
New helper functionality